cmd/traffic/cmd/agent/agent.go

```diff
@@ -73,37 +73,61 @@ func AppEnvironment() map[string]string {
 	return fullEnv
 }
 
-// svcAccPath is the path where the ServiceAccount Admission Controller automatically provides its secrets.
-const svcAccPath = "/var/run/secrets/kubernetes.io"
 const tpMountsEnv = "TELEPRESENCE_MOUNTS"
 
-func (cfg *Config) HasMounts(ctx context.Context, env map[string]string) (bool, error) {
+func (cfg *Config) HasMounts(ctx context.Context, env map[string]string) bool {
 	tpMounts := env[tpMountsEnv]
-	stat, err := os.Stat(svcAccPath)
-	if err != nil || !stat.IsDir() {
-		return tpMounts != "", nil
+	retVal := false
+	split := strings.Split(tpMounts, ":")
+	for _, itemPath := range split {
+		dlog.Debugf(ctx, "agent mount path: %s", itemPath)
+		retVal = true
 	}
+	return retVal
+}
 
-	// This must be included in the shared mounts unless it's already provided
-	svcAccLink := filepath.Join(cfg.AppMounts, svcAccPath)
-	if stat, err = os.Stat(svcAccLink); err == nil && stat.IsDir() {
-		return true, nil
-	}
+// Add any token-rotating system secrets directories if they exist
+// e.g. /var/run/secrets/kubernetes.io or /var/run/secrets/eks.amazonaws.com
+func (cfg *Config) addSecretsMounts(ctx context.Context, env map[string]string) error {
+	var retErr error = nil
+	tpMounts := env[tpMountsEnv]
 
-	// Add a link to the kubernetes.io directory under {{.AppMounts}}/var/run/secrets
-	if err = os.MkdirAll(filepath.Dir(svcAccLink), 0700); err != nil {
-		return false, err
-	}
-	if err = os.Symlink(svcAccPath, svcAccLink); err != nil {
-		return false, err
-	}
-	if tpMounts == "" {
-		tpMounts = svcAccPath
-	} else {
-		tpMounts += ":" + svcAccPath
+	// This will attempt to handle all the secrets dirs, but will return the first error we encountered.
+	if secretsDir, retErr := os.Open("/var/run/secrets"); retErr == nil {
+		if fileInfo, retErr := secretsDir.Readdir(-1); retErr == nil {
+			secretsDir.Close()
+			for _, file := range fileInfo {
+				// Files that rotate, such as tokens found in /var/run/secrets, we need to symlink to the original
+				// (Stops at the first error found.)
+				if retErr == nil && file.IsDir() {
+					dirPath := "/var/run/secrets/" + file.Name()
+					dlog.Debugf(ctx, "checking agent secrets mount path: %s", dirPath)
+					if stat, retErr := os.Stat(dirPath); retErr == nil && stat.IsDir() {
```

Comment:
It's better to handle the unhappy path directly and provide some context to the error, e.g.
```
state, err := os.Stat(dirPath)
if err != nil {
	return fmt.Errorf("unable to stat %s: %w", err)
}
```